src/analyze/analyze-condition.c

```diff
@@ -72,18 +73,35 @@ static int log_helper(void *userdata, int level, int error, const char *file, in
         return r;
 }
 
-int verify_conditions(char **lines, UnitFileScope scope) {
+int verify_conditions(char **lines, UnitFileScope scope, const char *root) {
         _cleanup_(manager_freep) Manager *m = NULL;
+        _cleanup_free_ char *var = NULL;
+        _cleanup_free_ char **filenames = NULL;
         Unit *u;
         char **line;
         int r, q = 1;
 
+        STRV_FOREACH(line, lines)
+                if (unit_name_is_valid(*line, UNIT_NAME_ANY)) {
+                        r = strv_push(&filenames, *line);
```

Comment:
the string *line is now part of two strv, that's a big ugly.

might be OK to do, but really needs a comment at the very least, ad the declartion of the variable. but quite frankly, it's not worth it, this stuff is not sensitive to performance, such an optimization sounds unnecessary and is needleslly confusing hence.

Reply:
Replaced with `strv_extend` so we copy the string instead.